In [1]:
#import dataset
import pandas as pd
import os
sale_df = pd.read_csv(r'D:\DS_DataSpark Illuminating Insights for Global Electronics\raw_datasets\Sales.csv')

In [2]:
sale_df.head(3)

Order Number  Line Item Order Date Delivery Date  CustomerKey  StoreKey  \
0        366000          1   1/1/2016           NaN       265598        10   
1        366001          1   1/1/2016     1/13/2016      1269051         0   
2        366001          2   1/1/2016     1/13/2016      1269051         0   

   ProductKey  Quantity Currency Code  
0        1304         1           CAD  
1        1048         2           USD  
2        2007         1           USD

check for null values

In [3]:
sale_df.isnull().sum()

Order Number         0
Line Item            0
Order Date           0
Delivery Date    49719
CustomerKey          0
StoreKey             0
ProductKey           0
Quantity             0
Currency Code        0
dtype: int64

drop delivery date column as it contains 49719 null records

In [4]:
sale_df = sale_df.drop(columns=['Delivery Date'])

check for duplicates

In [5]:
sale_df.duplicated().sum()

np.int64(0)

data type conversion

In [6]:
sale_df.dtypes

Order Number      int64
Line Item         int64
Order Date       object
CustomerKey       int64
StoreKey          int64
ProductKey        int64
Quantity          int64
Currency Code    object
dtype: object

In [7]:
sale_df['Order Date'] = pd.to_datetime(sale_df['Order Date'],format='%m/%d/%Y')

standardize text data

In [8]:
sale_df['Currency Code'] = sale_df['Currency Code'].str.strip().str.upper()

Rename columns

In [9]:
sale_df.columns

Index(['Order Number', 'Line Item', 'Order Date', 'CustomerKey', 'StoreKey',
       'ProductKey', 'Quantity', 'Currency Code'],
      dtype='object')

In [10]:
sale_df = sale_df.rename(columns={
    'Order Number': 'order_number',
    'Line Item': 'line_item',
    'Order Date': 'order_date',
    'CustomerKey': 'customer_key',
    'StoreKey': 'store_key',
    'ProductKey': 'product_key',
    'Quantity': 'quantity',
    'Currency Code': 'currency_code',
})

check stats

In [11]:
sale_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62884 entries, 0 to 62883
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   order_number   62884 non-null  int64         
 1   line_item      62884 non-null  int64         
 2   order_date     62884 non-null  datetime64[ns]
 3   customer_key   62884 non-null  int64         
 4   store_key      62884 non-null  int64         
 5   product_key    62884 non-null  int64         
 6   quantity       62884 non-null  int64         
 7   currency_code  62884 non-null  object        
dtypes: datetime64[ns](1), int64(6), object(1)
memory usage: 3.8+ MB


In [12]:
sale_df.describe()

order_number     line_item                     order_date  \
count  6.288400e+04  62884.000000                          62884   
mean   1.430905e+06      2.164207  2018-11-30 21:21:56.939126016   
min    3.660000e+05      1.000000            2016-01-01 00:00:00   
25%    1.121017e+06      1.000000            2018-01-25 00:00:00   
50%    1.498016e+06      2.000000            2019-02-06 00:00:00   
75%    1.788010e+06      3.000000            2019-11-23 00:00:00   
max    2.243032e+06      7.000000            2021-02-20 00:00:00   
std    4.532963e+05      1.365170                            NaN   

       customer_key     store_key   product_key      quantity  
count  6.288400e+04  62884.000000  62884.000000  62884.000000  
mean   1.180797e+06     31.802144   1125.859344      3.144790  
min    3.010000e+02      0.000000      1.000000      1.000000  
25%    6.808580e+05      8.000000    437.000000      1.000000  
50%    1.261200e+06     37.000000   1358.000000      2.000000  
75%    1.686496e+06     53.000000   1650.000000      4.000000  
max    2.099937e+06     66.000000   2517.000000     10.000000  
std    5.859634e+05     22.978188    709.244010      2.256371

Export cleaned dataset

In [13]:
sale_df.to_csv('cleaned_dataset/sales.csv',index=False)

check dataset

In [14]:
sale_df

order_number  line_item order_date  customer_key  store_key  \
0            366000          1 2016-01-01        265598         10   
1            366001          1 2016-01-01       1269051          0   
2            366001          2 2016-01-01       1269051          0   
3            366002          1 2016-01-01        266019          0   
4            366002          2 2016-01-01        266019          0   
...             ...        ...        ...           ...        ...   
62879       2243030          1 2021-02-20       1216913         43   
62880       2243031          1 2021-02-20        511229          0   
62881       2243032          1 2021-02-20        331277          0   
62882       2243032          2 2021-02-20        331277          0   
62883       2243032          3 2021-02-20        331277          0   

       product_key  quantity currency_code  
0             1304         1           CAD  
1             1048         2           USD  
2             2007         1           USD  
3             1106         7           CAD  
4              373         1           CAD  
...            ...       ...           ...  
62879          632         3           USD  
62880           98         4           EUR  
62881         1613         2           CAD  
62882         1717         2           CAD  
62883          464         7           CAD  

[62884 rows x 8 columns]